### PR TITLE
Small clean up of completion code

### PR DESCRIPTION
**Pull request by bthesorceror**

### CODE REVIEW

File: src/visitors/semantic/completion/mod.rs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        }
+        _ => None,
+    }
+}
+
+impl<'a> Visitor<'a> for CompletableFinderVisitor {
+    fn visit(&mut self, node: Rc<Node<'a>>) -> bool {
+        let mut state = self.state.borrow_mut();
+        let loc = node.loc();
+        let pos = self.pos.clone();
```
Comment (kelwang) [inline src/visitors/semantic/completion/mod.rs]: you can borrow here instead of copy

File: src/visitors/semantic/completion/mod.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            || (loc.start.line == pos.line + 1
+                && loc.start.column > pos.character + 1)
+        {
+            return true;
+        }
+
+        if let Node::VariableAssgn(assgn) = node.as_ref() {
+            let id = assgn.id.clone();
+
+            if let Some(var_type) = get_var_type(assgn.init.clone()) {
```
Comment (kelwang) [inline src/visitors/semantic/completion/mod.rs]: borrow here

File: src/visitors/semantic/completion/mod.rs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+#[derive(Clone)]
+struct VarResult {
+    pub name: String,
+    pub var_type: VarType,
+}
+
+impl VarResult {
+    pub fn detail(&self) -> String {
```
Comment (kelwang) [inline src/visitors/semantic/completion/mod.rs]: It's cleaner to implement a Display trait for the VarType.

File: src/visitors/semantic/completion/mod.rs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+        CompletableFinderVisitor {
+            state: Rc::new(RefCell::new(
+                CompletableFinderState::default(),
+            )),
+            pos,
+        }
+    }
+}
+
+fn get_var_type(expr: Expression) -> Option<VarType> {
```
Comment (kelwang) [inline src/visitors/semantic/completion/mod.rs]: you can also borrow here

`fn get_var_type(expr: &Expression) -> Option<VarType> {`

File: src/visitors/semantic/utils.rs [hunk 5]
```diff
@@ -89,3 +89,13 @@ pub fn map_node_to_location(uri: String, node: Rc<Node>) -> Location {
         },
     }
 }
+
+pub fn follow_function_pipes(c: Box<CallExpr>) -> MonoType {
```
Comment (kelwang) [inline src/visitors/semantic/utils.rs]: change this to 
```rust
pub fn follow_function_pipes(c: &Box<CallExpr>) -> &MonoType {
    if let Some(p) = &c.pipe {
        if let Expression::Call(call) = p {
            return follow_function_pipes(&call);
        }
    }

    &c.typ
}
```

File: src/visitors/semantic/completion/mod.rs (hunk 3 above)
Comment (bthesorceror) [reply src/visitors/semantic/completion/mod.rs]: this is directly related to a field we return to the LSP client named detail, so I prefer to be explicit here.

File: src/visitors/semantic/completion/mod.rs (hunk 2 above)
Comment (bthesorceror) [reply src/visitors/semantic/completion/mod.rs]: Ownership with these visitors using walk in libflux can get tricky, prefer to clone than untangle ownership

File: src/visitors/semantic/completion/mod.rs (hunk 4 above)
Comment (bthesorceror) [reply src/visitors/semantic/completion/mod.rs]: Ownership with these visitors using walk in libflux can get tricky, prefer to clone than untangle ownership

File: src/visitors/semantic/utils.rs (hunk 5 above)
Comment (bthesorceror) [reply src/visitors/semantic/utils.rs]: Ownership with these visitors using walk in libflux can get tricky, prefer to clone than untangle ownership

File: src/visitors/semantic/completion/mod.rs (hunk 1 above)
Comment (bthesorceror) [reply src/visitors/semantic/completion/mod.rs]: Ownership with these visitors using walk in libflux can get tricky, prefer to clone than untangle ownership

**Pull request merged**